Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -135,6 +135,41 @@ open class NavigationMapView: UIView {
         }
     }
     
+    var customizedPuckType: PuckType? {
```

Comment:
I wasn't following initial discussion regarding new puck functionality, but if our aim is to provide both 2D/3D puck capabilities and retain already existing `UserCourseView` what about exposing such entity:
```swift
public enum PuckType {
    
    case puck2D(configuration: Puck2DConfiguration)
    
    case puck3D(configuration: Puck3DConfiguration)
    
    case `default`(userCourseView: UserCourseView)
}
```

Based on such API user will be able to provide 2D/3D implementation of the puck whereas Navigation SDK will be providing default implementation like this:

```swift
public var puckType: PuckType = .default(userCourseView: UserPuckCourseView(frame: CGRect(origin: .zero, size: 75.0)))
```

/cc @1ec5

Reply:
Thanks for the suggestion! I'll improve this one by your advice!